### PR TITLE
Added documentation for mirroring feature

### PR DESCRIPTION
**Pull request by kandelvijaya**
Based on the proposal which can be [found here ](https://github.com/apple/swift-evolution/blob/master/proposals/0219-package-manager-dependency-mirroring.md ) and the implementation [found here](https://github.com/apple/swift-package-manager/pull/1776) I have added documentation for the mirroring feature. Thanks, @aciidb0mb3r

### CODE REVIEW

File: Documentation/Usage.md [hunk 1]
```diff
@@ -393,6 +393,31 @@ import Foundation
 #endif
 ```
 
+## Using Mirrored packages
```
Comment (aciidb0mb3r) [inline Documentation/Usage.md]: I think we should just say:

```
## Dependency Mirroring
```

File: Documentation/Usage.md [hunk 2]
```diff
@@ -393,6 +393,31 @@ import Foundation
...
+
+To add mirror for a given package:
```
Comment (aciidb0mb3r) [inline Documentation/Usage.md]: I think it'll good to add some description of the feature before diving into an example. It'll also be nice to briefly mention the config file.

File: Documentation/Usage.md [hunk 3]
```diff
@@ -393,6 +393,31 @@ import Foundation
...
+```sh
+$ swift package config set-mirror --package-url https://github.com/apple/swift-nio.git --mirror-url ../LocalDeps/nio
```
Comment (aciidb0mb3r) [inline Documentation/Usage.md]: One of the primary use-case of this feature is mirroring the repository on some other server. I think it'll be better to use mirror-url which is an actual url. Maybe something like this:

```sh
swift package config set-mirror --package-url https://github.com/apple/swift-nio.git --mirror-url https://github.com/<your-org>/swift-nio.git
```

File: Documentation/Usage.md (hunk 2 above)
Comment (kandelvijaya) [reply Documentation/Usage.md]: Sounds good. Let me draft a brief description.

Comment (kandelvijaya) [reply Documentation/Usage.md]: done.

File: Documentation/Usage.md [hunk 4]
```diff
@@ -393,6 +393,49 @@ import Foundation
 #endif
 ```
 
+## Dependency Mirroring
+
+A dependency mirror refers to an alternate source location which exactly replicates the contents of the original source. Mirrors can ensure availability even if original source is moved/unavailable or deleted. Mirror information is stored in a local file at `<PackageRoot>/.swiftpm/config` in JSON format, however there are commands which allow you to add/delete without having to edit the file manually.
```
Comment (jakepetroules) [inline Documentation/Usage.md]: > which **exactly replicates** the contents of the original source

That's not strictly true, as you can fork a package for the purpose of making changes to it. I'd leave out the "exactly replicates" bit. How about:

"A dependency mirror refers to an alternate source location where a package is made available. Mirrors can ensure availability even if the original location is unavailable, or has been moved or deleted. [...]"

**Comment by aciidb0mb3r** [comment]
Closing due to inactivity, feel free to update and reopen.

**Pull request closed without merging**